tests/rptest/tests/datalake/datalake_e2e_test.py

```diff
@@ -44,3 +77,49 @@ def test_e2e_basic(self, query_engine, filesystem_catalog_mode):
             dl.create_iceberg_enabled_topic(self.topic_name)
             dl.produce_to_topic(self.topic_name, 1024, count)
             dl.wait_for_translation(self.topic_name, msg_count=count)
+
+    @cluster(num_nodes=4)
+    @matrix(query_engine=[QueryEngineType.SPARK, QueryEngineType.TRINO])
+    def test_avro_schema(self, query_engine):
+        count = 100
+        table_name = f"redpanda.{self.topic_name}"
+
+        with DatalakeServices(self.test_ctx,
+                              redpanda=self.redpanda,
+                              filesystem_catalog_mode=True,
+                              include_query_engines=[query_engine]) as dl:
+            dl.create_iceberg_enabled_topic(self.topic_name)
+            avro_serde_client = self._get_serde_client(SchemaType.AVRO,
+                                                       SerdeClientType.Golang,
+                                                       self.topic_name, count)
+            avro_serde_client.start()
+            avro_serde_client.wait()
```

Comment:
does this need any explicit cleanup like stop() or some such?

Reply:
It doesn't look like this is implemented for SerdeClient -- seems like the expectation is that the process exits after it finishes producing